include/swift/AST/Expr.h

```diff
@@ -5128,6 +5133,15 @@ inline bool Expr::isInfixOperator() const {
   return isa<BinaryExpr>(this) || isa<IfExpr>(this) ||
          isa<AssignExpr>(this) || isa<ExplicitCastExpr>(this);
 }
+  
+inline bool ApplyExpr::validateArg(Expr *e) const {
+  if (isa<SelfApplyExpr>(this))
+    return true;
+  else if (isa<BinaryExpr>(this))
+    return isa<TupleExpr>(e);
+  else
+    return isa<ParenExpr>(e) || isa<TupleExpr>(e) || isa<TupleShuffleExpr>(e);
```

Comment:
Is it worth to add `PrefixUnaryExpr`/`PostfixUnaryExpr` -> `ParenExpr` check here? cc: @xedin

Reply:
Since arguments of such expressions are _now_ implicitly wrapped into parens it shouldn't be required.

Reply:
That's why I thought we *can* add the check for it.

Reply:
I don’t think we should. To be honest, I’d like to drop the BinaryExpr -> TupleExpr check too, because I want to support operator functions with extra defaulted parameters (for #line and friends). But that will require other changes to the compiler, and I’d rather do one thing at a time.